akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -334,7 +334,7 @@ private[stream] object ConnectionSourceStage {
         connection ! ResumeReading
       }
 
-      override def onDownstreamFinish(): Unit = {
+      override def onDownstreamFinish(cause: Throwable): Unit = {
```

Comment:
Same here, connection abortion is a pretty hard side-effect, so might make sense to log the cause.